configure.ac

```diff
@@ -183,6 +183,11 @@ AC_ARG_WITH([zshcompletiondir],
 	[with_zshcompletiondir=${datadir}/zsh/site-functions])
 AC_SUBST([zshcompletiondir], [$with_zshcompletiondir])
 
+# Never enabled with autotools - let distros switch to meson if feature is wanted
+AC_DEFINE([ENABLE_DLOPEN_ZSTD], [0], [Enable dlopen zstd])
+AC_DEFINE([ENABLE_DLOPEN_XZ], [0], [Enable dlopen xz])
+AC_DEFINE([ENABLE_DLOPEN_ZLIB], [0], [Enable dlopen zlib])
```

Comment:
What is the reason behind the individual toggles - can we make it a single "use-dlopen-for-all" or alternatively indiscriminately switch to `dlopen` and remove the linking all together?

The systemd team have introduced a nice custom ELF section using which one can describe the (indirect) link dependency. IIRC tools like dracut, mkinitcpio, package managers have been (or in the process of) being update to make use of it.

I can open a follow-up PR adding the ELF section, if that's the blocker for us to always use `dlopen`

Reply:
The ELF section is in a wip commit on top of these... didn't send that yet as I had to do other things before finalizing it.

My intention is to allow distros to still link directly. At least for a few versions... the ELF section thing is too new to rely on and the whole dlopen thing doesn't work if you statically link  (statically linking was disabled in autotools for other reasons, but I know some downstreams have/had patches to workaround the restriction).

Adding a -Ddlopen=all-enabled seems reasonable.